flink-yarn/src/main/java/org/apache/flink/yarn/entrypoint/YarnEntrypointUtils.java

```diff
@@ -90,16 +90,27 @@ public static Configuration loadConfiguration(String workingDirectory, Map<Strin
 
 		final String keytabPath;
 
-		if (env.get(YarnConfigKeys.KEYTAB_PATH) == null) {
+		if (env.get(YarnConfigKeys.LOCAL_KEYTAB_PATH) == null) { // keytab not exist
 			keytabPath = null;
 		} else {
-			File f = new File(workingDirectory, Utils.KEYTAB_FILE_NAME);
-			keytabPath = f.getAbsolutePath();
+			File f;
+			f = new File(env.get(YarnConfigKeys.LOCAL_KEYTAB_PATH));
+			if (f.exists()) { // keytab file exist in host environment.
+				keytabPath = f.getAbsolutePath();
+			} else {
+				f = new File(workingDirectory, env.get(YarnConfigKeys.LOCAL_KEYTAB_PATH));
+				if (f.exists()) { // keytab file exist in working directory.
+					keytabPath = f.getAbsolutePath();
+				} else { // fall back to default keytab file
+					f = new File(workingDirectory, Utils.DEFAULT_KEYTAB_FILE);
```

Comment:
Aren't we duplicating the default logic here? If nothing is set the default setting in the Flink `Configuration` should already be `DEFAULT_KEYTAB_FILE`.

Reply:
yes. good catch. this piece of code was there before we use the YarnConfigOption approach. I would refine this.